src/Microsoft.Crank.AzureDevOpsWorker/RetryHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Crank.AzureDevOpsWorker
+{
+    internal class RetryHandler : DelegatingHandler
+    {
+        private const int MaxRetries = 3;
+
+        public RetryHandler(HttpMessageHandler innerHandler)
+            : base(innerHandler)
+        { }
+
+        protected override async Task<HttpResponseMessage> SendAsync(
+            HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            HttpResponseMessage response = null;
+            for (int i = 0; i < MaxRetries; i++)
+            {
+                response = await base.SendAsync(request, cancellationToken);
```

Comment:
Should there be any delay and/or logging between attempts?

Reply:
Thanks, done

Reply:
`HttpResponseMessage` should probably be disposed if it's not being returned?